index.bs

```diff
@@ -2206,18 +2221,23 @@ To <dfn>serialize an Array-like</dfn> given |production|, |handle id|, |known ob
 </div>
 
 <div algorithm>
-To <dfn>serialize as a list</dfn> given |iterable|, |max depth|, |ownership type|,
+To <dfn>serialize as a list</dfn> given |iterable|, |serialization options|, |ownership type|,
 |serialization internal map|, |realm|, and |session|:
 
   1. Let |serialized| be a new list.
 
   1. For each |child value| in |iterable|:
 
-    1. Let |child depth| be |max depth| - 1 if |max depth| is not null, or null
-       otherwise.
+    1. Let |child serialization options| be a [=map/clone=] of
+       |serialization options|.
+
+    1. If |child serialization options|["<code>maxObjectDepth</code>"] is
+       not null, set |child serialization
+       options|["<code>maxObjectDepth</code>"] to |child serialization
+       options|["<code>maxObjectDepth</code>"] - 1.
```

Comment:
I'm fairly sure we can move the cloning out of the loop so that we do not have to do it for each and every child. Also when should happen when this method gets a `maxObjectDepth` of `0` passed in? Right now we end-up again with `-1`.